### DISCUSSION TITLE
Providing Hub APIs over custom domains

### DISCUSSION THREAD

**Discussion by andrewxhill** [textileio/community#294] [Hub]
We've had a number of projects request things between custom domains for Hub APIs and full white labelled system. The primary one I have heard is getting Bucket Links (IPFS, IPNS, and Thread links) over a custom domain pointed to the Hub infra. e.g. 

```
▶ hub buck links
> Your bucket links:
> https://{custom.domain.com}/thread/bafkwh5jwn7l76nxmmirel57ojkulaphnuecro3cmex4nibqhhseysii/buckets/bafzbeidm2siabgz4imyuk5lldy4wyo7iyxblzzuwbl2ndxucp5nuhsjcly Thread link
> https://{custom.domain.com}/ipns/bafzbeidm2siabgz4imyuk5lldy4wyo7iyxblzzuwbl2ndxucp5nuhsjcly IPNS link
> https://{custom.domain.com}/ipfs/bafybeihh3iaysmb7jbrzdxtz2htguh3wl76kafu7ipuqgzskvk2wuf7kwq IPFS link
```

If your project is interested in these features, let us know how you would like to use this so that we can build it right!

**Comment by georgemac510** [comment]
Hi Andrew,

Please include me in this!

Hub is working great right now. Seems like it is one of a few things
working well right now in the Filecoin ecosystem.

For my purposes, it would be great if the IPNS links had a shelf life
longer than just a few days for a start.

If this is not doable, can we get some guidance on what the index.html
would look like to get the textile.space link to work in the place of the
IPNS link? I bet it’s something pretty simple, but I may be looking at the
wrong thing or in the wrong place.

Let me know how I can help.

Hub is working out very well. Great job by the team!

Best,

John McLaughlin
jmclaughlin@mckurzcapital.com
georgemac510@gmail.com

**Comment by shivkumarsingh7** [comment]
Hello Andrew,

Is _custom.domain.com_ feature is implemented or still in discussion?

**Reply by sanderpick** [reply]
hey @shivkumarsingh7, we still don't have this feature, mostly because it is possible to simply CNAME you custom domain the `https://bucket_key.textile.space` (bucket key is the same as IPNS key).
